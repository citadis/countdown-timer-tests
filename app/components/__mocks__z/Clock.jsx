'use strict'

const Clock = jest.genMockFromModule('../Clock')

function formatTime (timeInSeconds) {
  return '1001'
}

function render () {
  let { timeInSeconds } = Clock.props
  return (
    <div className='clock'>
      <span className='clock-text'>
        {this.formatTime(timeInSeconds)}
      </span>
    </div>
  )
}

Clock.formatTime = formatTime
Clock.render = render

module.exports = Clock
