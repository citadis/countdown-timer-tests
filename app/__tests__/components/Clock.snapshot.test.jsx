import React from 'react'
import renderer from 'react-test-renderer'

import Clock from 'Clock'

describe('Clock component renders the clock correctly', () => {
  it('renders correctly', () => {
    const seconds = 64
    const rendered = renderer.create(
      <Clock timeInSeconds={seconds} />
    )

    expect(rendered.toJSON()).toMatchSnapshot()
  })

  it('renders correctly with less than 10 seconds on the clock', () => {
    const seconds = 0
    const rendered = renderer.create(
      <Clock timeInSeconds={seconds} />
    )

    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
