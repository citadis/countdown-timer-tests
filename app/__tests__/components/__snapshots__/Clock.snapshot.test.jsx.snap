// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clock component renders the clock correctly renders correctly 1`] = `
<div
  className="clock"
>
  <span
    className="clock-text"
  >
    01:04
  </span>
</div>
`;

exports[`Clock component renders the clock correctly renders correctly with less than 10 seconds on the clock 1`] = `
<div
  className="clock"
>
  <span
    className="clock-text"
  >
    00:00
  </span>
</div>
`;
